tasks/resolve_reference.rb

```diff
@@ -32,77 +35,41 @@ def config_client(opts)
     Aws::EC2::Client.new(options)
   end
 
-  def resolve_reference(opts)
-    client = config_client(opts)
+  def resolve_reference(config, template)
+    unless template.key?(:uri) || template.key?(:name)
+      msg = "You must provide a 'name' or 'uri' in 'target_mapping' for the AWS plugin"
```

Comment:
Could we do this validation with the task metadata?

Reply:
I don't think so - both are technically optional because you can provide one or the other, but you can't provide neither. I'm not sure that we can express that in task metadata?

Reply:
Oh right, its not json schema.